### ISSUE TITLE
setDatesDisabled not working

### ISSUE THREAD

**Issue by RainenterMudin** [uxsolutions/bootstrap-datepicker#2661]
### Expected behaviour
Tell us what should happen.
dates on array disabled
### Actual behaviour
Tell us what happens instead.
datss on array not disabled
### Datepicker version used

ex. 1.6.1.

This is the datepicker init
$('#date').datepicker({
                        // format: "yyyy/dd/mm",
                        format: "yyyy-mm-dd",
                        startDate: "1d",
                        endDate: "+30d",
                        // todayBtn: "linked",
                        language: "id",
                        orientation: "top right",
                        keyboardNavigation: false,
                        autoclose: true,
                        // daysOfWeekDisabled: disabledWeekDays,
                        beforeShowDay: function(date) {
                            var allDates = date.getFullYear() + '-' + (date.getMonth() + 1) + '-' + date.getDate();
                            if (dissableddateee2.indexOf(allDates) != -1)
                                return false;
                            else
                                return true;

                        }
                    });

this is 
$('#date').datepicker('setDatesDisabled', ['2022-08-15']); --> array just example
### Example code

Jsfiddle example to reproduce the problem.

**Comment by KevinChauvet** [comment]
Same, I'm trying to find some information about it, but for now it's just not working.
It's working on the older version of datepicker, but if you use datepicker3 it's not.

**Comment by ptkato** [comment]
I think it has to do with the `startDate` and `endDate` options, when I drop those, `datesDisabled` works.

**Comment by KevinChauvet** [comment]
Indeed, if I remove them it works as well for me. But on the documentation, it's written that they are still actual

**Comment by ptkato** [comment]
Actually, let me rectify that, it seems to work while `startDate`/`endDate` are set, however it is dependent on the `format` option, if I feed it something like `2023-12-13`, the format must be `yyyy-mm-dd`, otherwise `datesDisabled` won't work.
